src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -355,13 +355,29 @@ public string DownloadFile(string storageKeyOfFile)
 		/// <param name="storageKeyOfBookFolder"></param>
 		public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinationParentDirectory, ProgressDialog downloadProgress = null)
 		{
-			//TODO tell it not to download pdfs. Those are just in there for previewing purposes, we don't need to get them now that we're getting the real thing
-
 			//review: should we instead save to a newly created folder so that we don't have to worry about the
 			//other folder existing already? Todo: add a test for that first.
 
-			if (!GetBookExists(storageKeyOfBookFolder))
+			// We need to download individual files to avoid downloading unwanted files (PDFs and thumbs.db to
+			// be specific).  See https://silbloom.myjetbrains.com/youtrack/issue/BL-2312.  So we need the list
+			// of items, not just the count.
+			var matching = GetMatchingItems(storageKeyOfBookFolder);
+			int totalItems = 0;
+			string folderName = null;
+			for (int i = 0; i < matching.S3Objects.Count; ++i)
+			{
+				var objKey = matching.S3Objects[i].Key;
+				if (String.IsNullOrEmpty(folderName))
```

Comment:
Is there an assumption here that the directory is flat? I can't tell. Although all existing books on bl.org are flat as far as I know, Bloom 3.3 and following allows books to have a "template" folder full of thumbnails (see Basic Book). So going forward, we can't assume flatness.